django/core/serializers/python.py

```diff
@@ -70,15 +70,22 @@ def handle_m2m_field(self, obj, field):
                 def m2m_value(value):
                     return value.natural_key()
 
+                def queryset_iterator(obj, field):
+                    return getattr(obj, field.name).iterator()
+
             else:
 
                 def m2m_value(value):
                     return self._value_from_field(value, value._meta.pk)
 
+                def queryset_iterator(obj, field):
+                    return getattr(obj, field.name).only("pk").iterator()
+
             m2m_iter = getattr(obj, "_prefetched_objects_cache", {}).get(
                 field.name,
-                getattr(obj, field.name).iterator(),
+                queryset_iterator(obj, field),
```

Comment:
We should add the same to the `django.core.serializers.xml_serializer.Serializer.handle_m2m_field`.